test/cluster-setup.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+# Copyright 2018 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script calls out to scripts in knative/test-infra to setup a cluster
+# and deploy the Pipeline CRD to it for running integration tests.
+
+source $(dirname $0)/../vendor/github.com/knative/test-infra/scripts/e2e-tests.sh
+
+set -o xtrace
+set -o errexit
+set -o pipefail
+
+function take_down_pipeline() {
+    header "Tearing down Pipeline CRD"
+    ko delete --ignore-not-found=true -f config/
+}
+trap take_down_pipeline EXIT
+
+# Called by `fail_test` (provided by `e2e-tests.sh`) to dump info on test failure
+function dump_extra_cluster_state() {
+  for crd in pipelines pipelineruns tasks taskruns resources pipelineparams
+  do
+    echo ">>> $crd:"
+    kubectl get $crd -o yaml --all-namespaces
+  done
+  echo ">>> Pipeline controller log:"
+  kubectl -n knative-build-pipeline logs $(kubectl -n knative-build-pipeline get pods -l app=build-pipeline-controller -o name)
+  echo ">>> Pipeline webhook log:"
+  kubectl -n knative-build-pipeline logs $(kubectl -n knative-build-pipeline get pods -l app=build-pipeline-webhook -o name)
+}
+
+set +o xtrace
```

Comment:
did the kitten die of delight b/c it was so happy to be able to see lines as they execute? :D

maybe you know a better way to handle this? I'd like to output the commands as I run them, but then it's hard b/c the functions I'm calling also do that, or alternatively don't work well when I do that, so it's hard to handle

Reply:
When everything works then it's nice not to have the clutter introduced by xtrace, however it's been experience that things often go wrong, and it's very hard to debug bash without knowing what's actually getting executed.

Even in the context of this PR, things have gone wrong, and it's been hard to understand why because I can't see everything that's happening.

And even now I have really no idea how most of this works, and I can't find what's actually being called (e.g. what's calling boskos? what's the mechanism that's looking for the `e2e-tests.sh` script and calling it with `--run-tests`? where does `DOCKER_REPO_OVERRIDE` get set?)